code/modules/mob/living/simple_animal/abnormality/he/red_shoes.dm

```diff
@@ -0,0 +1,415 @@
+	if (!shoes)//No runtimes after suppression, woohoo!
+		return
+	if(shoes.status_flags & GODMODE)//don't get possessed if it's already breaching
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/desire_move)
+		blackboard[BB_INSANE_CURRENT_ATTACK_TARGET] = shoes
+
+/datum/ai_behavior/desire_move//define AI behavior
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM
+	var/list/current_path = list()
+
+/datum/ai_behavior/desire_move/perform(delta_time, datum/ai_controller/controller)//Paths the pancicked to red shoes, causes runtimes if it's dead
+	. = ..()
+	var/mob/living/carbon/human/living_pawn = controller.pawn//the panicked
+	if(IS_DEAD_OR_INCAP(living_pawn))//stop if the panicked is dead
+		return
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/target = controller.blackboard[BB_INSANE_CURRENT_ATTACK_TARGET]
+	if(!istype(target))
+		finish_action(controller, FALSE)
+		return
+	if(!LAZYLEN(current_path))
+		current_path = get_path_to(living_pawn, target, /turf/proc/Distance_cardinal, 0, 80)
+		if(!current_path) // Returned FALSE or null.
+			finish_action(controller, FALSE)
+			return
+	addtimer(CALLBACK(src, .proc/Movement, controller), 1.25 SECONDS, TIMER_UNIQUE)
+	if(isturf(target.loc) && living_pawn.Adjacent(target))
+		finish_action(controller, TRUE)
+		return
+
+/datum/ai_behavior/desire_move/proc/Movement(datum/ai_controller/controller)//Ditto
+	var/mob/living/carbon/human/living_pawn = controller.pawn
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/target = controller.blackboard[BB_INSANE_CURRENT_ATTACK_TARGET]
+	if(!target)//might fix runtimes? 4/22
+		return
+	if(!LAZYLEN(current_path))
+		return
+	var/target_turf = current_path[1]
+	step_towards(living_pawn, target_turf)
+	current_path.Cut(1, 2)
+	if(target)
+		if(isturf(target.loc) && living_pawn.Adjacent(target))
+			finish_action(controller, TRUE)
+			return
+
+/datum/ai_behavior/desire_move/finish_action(datum/ai_controller/controller, succeeded)//When the panicked reach Red Shoes
+	. = ..()
+	var/mob/living/carbon/human/living_pawn = controller.pawn
+	var/obj/item/held = living_pawn.get_active_held_item()
+	var/mob/living/simple_animal/hostile/abnormality/red_shoes/target = controller.blackboard[BB_INSANE_CURRENT_ATTACK_TARGET]
+	if(succeeded)
+		living_pawn.cut_overlay(mutable_appearance('icons/effects/effects.dmi', "suicide", -ABOVE_MOB_LAYER))//This is SHIT! do it better!
+		living_pawn.cut_overlay(mutable_appearance('icons/effects/effects.dmi', "wander", -ABOVE_MOB_LAYER))//it was suggested to replace this with a loop
+		living_pawn.cut_overlay(mutable_appearance('icons/effects/effects.dmi', "breach", -ABOVE_MOB_LAYER))
+		living_pawn.cut_overlay(mutable_appearance('icons/effects/32x64.dmi', "panicked", -ABOVE_MOB_LAYER))
+		living_pawn.dropItemToGround(held)
+		target.Assimilate(living_pawn)//breaches red shoes with target as the argument for user
+	controller.blackboard[BB_INSANE_CURRENT_ATTACK_TARGET] = null
```

Comment:
Move like ALL of this to the lower part of this document. Make this hard to read. In general, keep similarly pathed things together. i.e.
`/mob/living/simple_animal/hostile/abnormality/red_shoes/...`
`/datum/ai_behavior/desire_move/...`
`/datum/status_effect/red_possess/...`